src/Subscription/Subscriber/MetadataSubscriberAccessorRepository.php

```diff
@@ -31,10 +31,11 @@ final class MetadataSubscriberAccessorRepository implements SubscriberAccessorRe
     public function __construct(
         private readonly iterable $subscribers,
         private readonly SubscriberMetadataFactory $metadataFactory = new AttributeSubscriberMetadataFactory(),
-        array $argumentResolvers = [],
+        iterable $argumentResolvers = [],
```

Comment:
We would also need to update the type for `$argumentResolvers` in the doc-block.
```php

    /**
     * @param iterable<object> $subscribers
     * @param iterable<ArgumentResolver>|list<ArgumentResolver> $argumentResolvers
     */
```